torch/quantization/fake_quantize.py

```diff
@@ -25,31 +25,45 @@ def __init__(self, dtype=torch.quint8, qscheme=torch.per_tensor_affine,
         self.qscheme = qscheme
         self.quant_min = quant_min
         self.quant_max = quant_max
-        self.enabled = True
+        self.fake_quant_enabled = True
+        self.observer_enabled = True
         self.observer = default_observer(dtype=dtype, qscheme=qscheme)()
         self.scale = None
         self.zero_point = None
 
-    def enable(self, enabled=True):
-        self.enabled = enabled
+    def enable_fake_quant(self, enabled=True):
+        self.fake_quant_enabled = enabled
         return self
 
-    def disable(self):
-        return self.enable(False)
+    def disable_fake_quant(self):
+        return self.enable_fake_quant(False)
+
+    def enable_observer(self, enabled=True):
+        self.observer_enabled = enabled
+
+    def disable_observer(self):
+        return self.enable_observer(False)
 
     def calculate_qparams(self):
         return self.observer.calculate_qparams()
 
     def forward(self, X):
-        if self.enabled:
-            self.observer(X)
+        if self.observer_enabled:
+            X = self.observer(X)
             scale, zero_point = self.calculate_qparams()
```

Comment:
`observer_enabled` also means update `self.scale` and `self.zero_point`?

Reply:
Yes, thats the intent. In a later PR we can optimize this further to have a seperate flag for controlling scale/zero-point calculation